source/Optimization.rst

```diff
@@ -0,0 +1,37 @@
+Optimisation
+===============
+
+
+PerTickCache
+----------------
+This function is used to retrieve values from the cache if some value is already computed. If the value is not present in the cache, it computes the value using a user-defined function and stores it in the cache.
+This is extremely useful in reducing the simulation time when we repeatedly have to query the context to compute the value for every agent in a tick.
+
+For example, consider the case in the simple SIR model where we have to find the fraction of infected individuals around an agent in a particular place. Here the infected fraction of all agents in the same place would be the same. So while running the simulation, we have to compute the same value repeatedly for all agents in that place. We need to query the context for all agents during each tick, thereby increasing the simulation time and computational resource required.
+Here when we implement PerTickCache, the infected fraction will be computed for all places as and when needed during each tick, and that value will be stored in the cache to be used for all the agents in that place.
```

Comment:
Please describe in some detail how the value is stored. For example, the modeller has to define a "unique key", otherwise the the value that is stored will not be accurately stored. This is explained in the ``documentation.md`` file, but you will need to reword it to make it a little more accessible. 

Please give an example of this as well. For example, in the example you give below, suppose you used the ``uniqueKey`` as just being ``placeType``, what would happen?